src/node.cc

```diff
@@ -1197,6 +1224,18 @@ enum encoding ParseEncoding(Isolate* isolate,
   }
 }
 
+
+enum encoding ParseEncoding(Isolate* isolate,
+                            Handle<Value> encoding_v,
+                            enum encoding _default) {
+  if (!encoding_v->IsString())
+    return _default;
+
+  node::Utf8Value encoding(isolate, encoding_v);
+
+  return ParseEncoding(*encoding, _default);
```

Comment:
You can probably do a little better still by also passing `encoding->length()`.  Then you can filter on length first before you start comparing bytes.